File: lcof/go/cong-wei-dao-tou-da-yin-lian-biao-lcof.go
/*
 * @Author: FisherCloud 
 * @Date: 2020-02-26 20:06:15 
 * @Last Modified by: FisherCloud
 * @Last Modified time: 2020-02-26 20:37:44
 */
// https://leetcode-cn.com/problems/cong-wei-dao-tou-da-yin-lian-biao-lcof/

package main

import (
	"fmt"
)

type ListNode struct {
	Val int
	Next *ListNode
}

func reversePrint(head *ListNode) []int {
	var ans, tmp []int
	for head != nil {
		tmp = append(tmp, head.Val)
		head = head.Next
	}
	
	for i := (len(tmp) - 1); i >= 0; i-- {
		fmt.Println(i)
		ans = append(ans, tmp[i])
	}

	return ans
}

// func main() {
// 	// fmt.Println(reversePrint)
// }

File: lcof/rs/cong-wei-dao-tou-da-yin-lian-biao-lcof.rs
/*
 * @Author: FisherCloud 
 * @Date: 2020-02-26 19:56:24 
 * @Last Modified by: FisherCloud
 * @Last Modified time: 2020-02-26 20:03:52
 */
// https://leetcode-cn.com/problems/cong-wei-dao-tou-da-yin-lian-biao-lcof/

// Definition for singly-linked list.
#[derive(PartialEq, Eq, Clone, Debug)]
pub struct ListNode {
  pub val: i32,
  pub next: Option<Box<ListNode>>
}

impl ListNode {
  #[inline]
  fn new(val: i32) -> Self {
    ListNode {
      next: None,
      val
    }
  }
}
#[derive(Debug)]
struct Solution {
}

impl Solution {
    pub fn reverse_print(head: Option<Box<ListNode>>) -> Vec<i32> {
        let mut ans:Vec<i32>  = vec![];
        let mut headlist = head;
        while let Some(node) = headlist {
            ans.push(node.val);
            headlist = node.next;
        }
        ans.reverse();
        return ans;
    }
}

fn main() {
    
}
